### ISSUE TITLE
Bug: mux-video-react and mux-player-react cause errors and build failures in Next.js

### ISSUE THREAD

**Issue by ndimatteo** [muxinc/elements#342]
### Which Mux Elements/Packages has a bug?

1. `@mux/mux-video-react`
2. `@mux/mux-player-react`

### Which browser(s) are you using?

**Latest:** Chrome, Firefox, Safari

### Which operating system(s) are you using?

macOS Monterey - MacBook Pro (14-inch, 2021)

### How are you using Mux Elements?

npm, via `yarn add xxxxx@latest` in a Next.js application

### Describe the bug

After installing either of these plugins, various client-side errors began popping up, some even tracing to _other_ (untouched) packages:
1. `TypeError: Cannot read properties of undefined (reading 'bind')` traced to [SWR](https://github.com/vercel/swr). <br />_(bug incorrectly reported to SWR [here](https://github.com/vercel/swr/issues/2129) by others who also confirmed this happened when mux packages were installed)_

2. `ReferenceError: window is not defined` traced to Framer Motion <br /><img width="984" alt="image" src="https://user-images.githubusercontent.com/737188/186470163-47c19683-6c40-4444-be89-4a5853814293.png">

3. `ReferenceError: self is not defined` from `@mux/mux-video-react` (or`@mux/mux-player-react`) <br /><img width="851" alt="image" src="https://user-images.githubusercontent.com/737188/186469561-9f5ac406-6a5f-4633-aef2-c1e190231053.png">

## Steps To Reproduce

1. in a Next.js application: `yarn add @mux/mux-video-react@latest`
2. import into a custom component, here's mine for example:
```js
import React from 'react'
import MuxVideo from '@mux/mux-video-react'

const MuxLoop = ({ playbackId, ...rest }) => {
  if (!playbackId) return null

  return (
    <MuxVideo
      playbackId={playbackId}
      streamType="on-demand"
      loop={true}
      muted={true}
      autoPlay={true}
      {...rest}
    />
  )
}

export default MuxLoop
```
3. Use your new component on any page
4. Visit a page locally and you'll get errors

**Link to code example:** https://codesandbox.io/s/mux-next-js-bug-ivujwd?file=/pages/index.js
_**PLEASE NOTE:** for whatever reason you'll need to fully restart the server after commenting in/out the `<MuxLoop />` component. I had to open a new terminal and run `yarn dev` and preview on a different port to get it to take_

## The current behavior
Using either of the `@mux` elements components `mux-video-react` or `mux-player-react` results in various errors, some tracing to other, unrelated components.

## The expected behavior
Importing an `@mux` elements component shouldn't cause errors out of the box within a Next.js application when using popular packages like SWR or Framer Motion.

**Comment by gkatsev** [comment]
Thanks for the issue. I think point 3 is already solved. It comes from a mux-embed transitive dependency that got updated last night. Make sure that you're on version 4.13.1 of it (`yarn list --pattern mux-embed` or `npm ls mux-embed`). I have an codesandbox where I removed framer-motion, and it loads as expected (and the yarn.lock shows mux-embed 4.13.1): https://codesandbox.io/s/mux-next-js-bug-forked-bhu8rf?file=/pages/index.js

The other two, I think, are related to: https://github.com/muxinc/elements/issues/334
Since, if I remove mux-video from the page, framer-motion loads as expected. https://codesandbox.io/s/mux-next-js-bug-forked-j6etmb?file=/pages/index.js
